packages/astro/src/types/public/config.ts

```diff
@@ -172,6 +172,8 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 	ssr?: ViteSSROptions;
 }
 
+export type CspAlgorithm = 'SHA-256' | 'SHA-384' | 'SHA-512';
```

Comment:
What you could do to keep a single source of truth is:
- extract the algorithm schema from the config
- use z.infer for this type